### PR TITLE
Fixes: The UIHoverGestureRecognizer availability starts with iOS 13

### LINKED ISSUE #781: [Bug] The UIHoverGestureRecognizer availability starts with ios 13
### Description

The usage of xct:TouchEffect will crash on iOS version less than 13

### Basic Information

- XCT Version 1.0.1

### PR DESCRIPTION
**Pull request by AndreiMisiukevich**
### Description of Change ###
Added iOS version checks before using UIHoverGestureRecognizer in TouchEffect.

### Bugs Fixed ###
- Fixes #781

### API Changes ###
None

### Behavioral Changes ###
None

### PR Checklist ###
- [ ] Has tests (if omitted, state reason in description)
- [X] Has samples (if omitted, state reason in description)
- [X] Rebased on top of main at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/PlatformTouchEffect.ios.cs [hunk 1]
```diff
@@ -13,14 +13,16 @@ namespace Xamarin.CommunityToolkit.iOS.Effects
...
+		UIGestureRecognizer touchGesture;
 
-		UIHoverGestureRecognizer hoverGesture;
+		UIGestureRecognizer hoverGesture;
 
 		TouchEffect effect;
 
 		UIView View => Container ?? Control;
 
+		bool IsiOS13OrNewer => UIDevice.CurrentDevice.CheckSystemVersion(13, 0);
```
Comment (jfversluis) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/PlatformTouchEffect.ios.cs:24]: Should we move this to somewhere where everyone can use this? We probably need this more often?

I know Forms has these checks too, but I guess they're internal? Or do they have something for this?

Comment (AndreiMisiukevich) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/PlatformTouchEffect.ios.cs:24]: Agreed. I will move it to some helper class

```var x = XCT.IsiOS13OrNewer;```

Does it look good?

Comment (AndreiMisiukevich) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/PlatformTouchEffect.ios.cs]: Done

**Review by pictos — APPROVED** [review]
LGTM!

**Pull request merged**